autoload/go/cmd.vim

```diff
@@ -83,12 +83,28 @@ function! go#cmd#Run(bang, ...)
         exe 'make!'
     endif
 
-    cwindow
-    let errors = getqflist()
+    let qflist = getqflist()
```

Comment:
Can you put a document above this line and explain why we do the iterating. Something like: `Do not add non valid filenames to avoid opening an empty buffer. Checkout https://github.com/fatih/vim-go/issues/287 for the problem`